src/project.c

```diff
@@ -772,6 +772,53 @@ static gboolean update_config(const PropertyDialogElements *e, gboolean new_proj
 		}
 		g_free(locale_path);
 	}
+
+	/* create filename path if it doesn't exist */
+	if ((err_code = utils_is_file_writable(locale_filename)) != 0) {
+
+		/* only create path if filename ends with '.geany' */
+		char *dot = strrchr(locale_filename, '.');
+		if (dot != NULL && strcmp(dot, ".geany") == 0) {
+
+			char path_delimiter = '/';
+
+#ifdef G_OS_WIN32
+            path_delimiter = '\\';
+#endif
+
+			/* get last path_delimiter so we know the path */
+			int i;
+			int last_path_delimiter;
+			for (i = 0; locale_filename[i] != '\0'; ++i) {
+				if (locale_filename[i] == path_delimiter) {
+					last_path_delimiter = i;
+				}
+			}
+
+			/* get substring containing only the filename path */
+			gchar *filename_path = g_malloc((last_path_delimiter + 2) * sizeof(char));
+			strncpy(filename_path, locale_filename, last_path_delimiter + 1);
+			filename_path[last_path_delimiter + 1] = '\0';
+
+			gboolean create_dir;
+
+			create_dir = dialogs_show_question_full(NULL, GTK_STOCK_OK, GTK_STOCK_CANCEL,
+				_("Create the project's filename path directory?"),
+				_("The path \"%s\" does not exist."),
+				filename_path);
+
+			if (create_dir) {
+				err_code = utils_mkdir(filename_path, TRUE);
+				g_free(filename_path);
+				if (err_code != 0) {
+					SHOW_ERR1(_("Filename path could not be created (%s)."),
+						g_strerror(err_code));
+					return FALSE;
+				}
+			}
+		}
```

Comment:
It looks like `filename_path` is leaked when this brace is reached if the `if (create_dir)` branch above isn't taken.